### ISSUE TITLE
Add ability to reorder terrains

### ISSUE THREAD

**Issue by bjorn** [mapeditor/tiled#1603]
Currently, it is not possible to change the order of the terrains defined in a tileset.

There should be only one difficulty with this, which is that reordering terrains affects the terrain IDs stored on the tiles, which will have to be adjusted accordingly. But this is similar to when a terrain is removed, which is already implemented.

Reported [on the forum](http://discourse.mapeditor.org/t/reordering-terrain-and-tilesets/2437).

**Comment by eishiya** [comment]
It seems that the ability to reorder terrains was removed at some point, perhaps when they were merged with Wang tiles? It would be nice to have it back.

**Comment by bjorn** [comment]
You're right, the ability to reorder Wang colors (and Wang sets) was never implemented, so now this feature is missing again.

**Comment by UliAbo** [comment]
That would be quite helpful to have! Is this something complicated to be implemented? I would love to see that in, as I have lots of Terrains now in our Project.

**Comment by bjorn** [comment]
> That would be quite helpful to have! Is this something complicated to be implemented? I would love to see that in, as I have lots of Terrains now in our Project.

Unfortunately it's not entirely trivial, since a `WangId` stores terrain _indexes_, which will all need updating when we allow terrains to be reordered. Alternatively we could switch to assigning terrain IDs. Either way it's not too complicated either, so I could probably still fit this into 1.9.

**Comment by spiralhug** [comment]
Wanted to bump this to say that this is still an issue and one we'd love to have implemented for our project too!
